lib/molecule-dev-environment.js

```diff
@@ -46,22 +101,169 @@ export default {
 
+    if (
+      global.atom.config.get("molecule-dev-environment.freeTerminal") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.freeTerminal", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pinnedPlans") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pinnedPlans", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginEslint") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginEslint", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginNightwatch") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginNightwatch", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginFlowtype") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginFlowtype", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginGlup") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginGlup", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginJest") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginJest", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginNpm") === undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginNpm", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginReactNative") ===
+      undefined
+    ) {
+      global.atom.config.set(
+        "molecule-dev-environment.pluginReactNative",
+        true,
+      );
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginShell") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginShell", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.pluginTestcafe") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.pluginTestcafe", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.moleculeConsole") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.moleculeConsole", true);
+    }
+    if (
+      global.atom.config.get("molecule-dev-environment.moleculeLinter") ===
+      undefined
+    ) {
+      global.atom.config.set("molecule-dev-environment.moleculeLinter", true);
+    }
+
+    let pluginsTable = [];
     // Bind plugins. Path should be relative to /lib/ExecutionControlEpic/Plugins/
-    [
-      "Eslint/index.js",
-      "Nightwatch/index.js",
-      "Flowtype.js",
-      "Gulp.js",
-      "Jest.js",
-      "Npm.js",
-      "ReactNative.js",
-      "Shell.js",
-      "Testcafe.js",
-    ].forEach(tool => provideDevtool(store, DevToolsControllerInstance)(tool));
+
+    if (
```

Comment:
All these ifs should be condensed into one loop. Something like:

``` js
for (const pluginName of Object.keys(pluginRoots)) {
    if (global.atom.config.get(`molecule.${pluginName}`)) {
      pluginsTable.push(pluginRoots[pluginName]);
      store.dispatch(featureLoad(pluginName));
    }
}
```

Reply:
I especially think that the table creation should be done with map / filter and the dispatch in a for / forEach, separation of concerns, two separated block of codes